backend/src/api/ressources/exams/controller.js

```diff
@@ -44,27 +44,25 @@ const controller = {
   },
 
   async editExercice(req, res, next) {
+    const { exercice } = res.locals;
     if (req.body) {
       try {
-        const result = await ExerciceData.update(req.params.exerciceId, req.body);
-        if (!result) {
-          return res.status(404).send('Exercice not Found');
-        }
-        return res.status(200).json(result);
+        exercice.update(req.body);
+        exercice.save();
+        return res.status(200).json(req.body);
       } catch (error) {
         return next(error);
       }
     } return res.status(400).send('Bad Request...');
   },
 
   async editExam(req, res, next) {
+    const { exam } = res.locals;
     if (req.body) {
       try {
-        const result = await ExamData.update(req.params.examId, req.body);
-        if (!result) {
-          return res.status(404).send('Exam not Found');
-        }
-        return res.status(200).json(result);
+        exam.update(req.body);
+        await exam.save();
```

Comment:
AFAIK, exam.update() returns a promise. https://mongoosejs.com/docs/documents.html

So we should `await`

Reply:
as far as I tested, it works properly
document.update in mongoose is not asynchronous, as it does not perform any operation in the database. It only updates the document variable and checks for model validation.
The operation in the database is done with `exam.save()`
More info : https://mongoosejs.com/docs/api.html#document_Document-update